File: evidence/generated-systems/golang/fintech-on-golang/fintech-on-golang/internal/model/Screening.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Screening Declaration
//==============================================================
type Screening struct {
    gorm.Model
     Score                                                            string
    ScreenedAt                                                            time.Time
    KycProfileId         *uint
    KycProfile           *KYCProfile `gorm:"foreignKey:KycProfileId"`
     Alerts           []ComplianceAlert `gorm:"foreignKey:AlertsFromScreeningId"`
    ScreeningType                      ScreeningType
    Status                      ScreeningStatus

// parent associations as their child

}

File: evidence/generated-systems/django/fintech-on-django/fintechOnDjango/models/Screening.py
from django.db import models
from fintechOnDjango.models.ScreeningType import ScreeningType
from fintechOnDjango.models.ScreeningStatus import ScreeningStatus

#======================================================================
# 
# Encapsulates data for model Screening
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Screening Declaration
#======================================================================
class Screening (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	score = RiskScore
	screenedAt = DateTime
	kycProfile = models.ForeignKey('KYCProfile', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	alerts = models.ManyToManyField('ComplianceAlert',  blank=True, related_name='+')
	screeningType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in ScreeningType])
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in ScreeningStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.score
		str = str + self.screenedAt
		str = str + self.screeningType
		str = str + self.status
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Screening";
    
	def objectType(self):
		return "Screening";
